modules/secrets-store-vault.adoc

```diff
@@ -0,0 +1,383 @@
+----
++
+.Example output
+[source,terminal]
+----
+Success! Data written to: auth/kubernetes/config
+----
+
+.. Create a policy for the application by running the following command:
++
+[source,terminal]
+----
+$ oc exec -i vault-0 --namespace=vault -- vault policy write csi -<<EOF
+  path "secret/data/*" {
+  capabilities = ["read"]
+  }
+  EOF
+----
++
+.Example output
+[source,terminal]
+----
+Success! Uploaded policy: csi
+----
+
+.. Create an authentication role to access the application by running the following command:
++
+[source,terminal]
+----
+$ oc exec -i vault-0 --namespace=vault -- vault write auth/kubernetes/role/csi \
+  bound_service_account_names=default \
+  bound_service_account_namespaces=default,test-ns,negative-test-ns \
+  policies=csi \
+  ttl=20m
+----
++
+.Example output
+[source,terminal]
+----
+Success! Data written to: auth/kubernetes/role/csi
+----
+
+.. Verify that all of the `vault` pods are running properly by running the following command:
++
+[source,terminal]
+----
+$ oc get pods -n vault
+----
++
+.Example output
+[source,terminal]
+----
+NAME                       READY   STATUS    RESTARTS   AGE
+vault-0                    1/1     Running   0          43m
+vault-csi-provider-87rgw   2/2     Running   0          19m
+vault-csi-provider-bd6hp   2/2     Running   0          19m
+vault-csi-provider-smlv7   2/2     Running   0          19m
+----
+
+.. Verify that all of the `secrets-store-csi-driver` pods are running properly by running the following command:
++
+[source,terminal]
+----
+$ oc get pods -n openshift-cluster-csi-drivers | grep -E "secrets"
+----
++
+.Example output
+[source,terminal]
+----
+secrets-store-csi-driver-node-46d2g                  3/3     Running   0             45m
+secrets-store-csi-driver-node-d2jjn                  3/3     Running   0             45m
+secrets-store-csi-driver-node-drmt4                  3/3     Running   0             45m
+secrets-store-csi-driver-node-j2wlt                  3/3     Running   0             45m
+secrets-store-csi-driver-node-v9xv4                  3/3     Running   0             45m
+secrets-store-csi-driver-node-vlz28                  3/3     Running   0             45m
+secrets-store-csi-driver-operator-84bd699478-fpxrw   1/1     Running   0             47m
+----
+
+. Create a secret provider class to define your secrets store provider:
+
+.. Create a YAML file that defines the `SecretProviderClass` object:
++
+.Example `secret-provider-class-vault.yaml`
+[source,yaml]
+----
+apiVersion: secrets-store.csi.x-k8s.io/v1
+kind: SecretProviderClass
+metadata:
+  name: my-vault-provider                   <1>
+  namespace: my-namespace                   <2>
+spec:
+  provider: vault                           <3>
+  parameters:                               <4>
+    roleName: "csi"
+    vaultAddress: "http://vault.vault:8200"
+    objects:  |
+      - secretPath: "secret/data/example1"
```

Comment:
@ropatil010 Please note that I updated several of the example values:

* "foo" -> "example1"
* "bar" -> "testSecret1"

(We try to avoid using "foo" and "bar" in product docs when possible). Please double check that I replaced everywhere appropriately. 

I also removed the second example, but let me know if you think it's important to include both.

Reply:
Got it, thanks for info. 
yeah You have updated the values correctly.